src/all/hentaifox/src/eu/kanade/tachiyomi/extension/all/hentaifox/HentaiFox.kt

```diff
@@ -100,4 +114,116 @@ class HentaiFox(
             Filter.Header("HINT: Use double quote (\") for exact match"),
         ) + super.getFilterList().list,
     )
+
+    private val sidebarPath = "includes/sidebar.php"
+
+    private fun sidebarMangaSelector() = "div.item"
+
+    private fun Element.sidebarMangaTitle() =
+        selectFirst("img")?.attr("alt")
+
+    private fun Element.sidebarMangaUrl() =
+        selectFirst("a")?.attr("abs:href")
+
+    private fun Element.sidebarMangaThumbnail() =
+        selectFirst("img")?.imgAttr()
+
+    private fun fetchHomepage(): Request {
+        return GET(baseUrl, headers)
+    }
+
+    private fun homepageCsrfParser(document: Document): String {
+        return document.select("[name=csrf-token]").attr("content")
+    }
+
+    private fun setSidebarHeaders(csrfToken: String?): Headers {
+        if (csrfToken == null) {
+            return xhrHeaders
+        }
+        return xhrHeaders.newBuilder()
+            .add("X-Csrf-Token", csrfToken)
+            .build()
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        // Sidebar mangas should always override any other search, so they should appear first
+        // in the case/when and only propagate to super when a "normal" search is issued
+        val sortOrderFilter = filters.filterIsInstance<SortOrderFilter>().firstOrNull()
+
+        return when {
+            sortOrderFilter?.state == 2 ->
+                sidebarRequest("top_rated")
+
+            sortOrderFilter?.state == 3 ->
+                sidebarRequest("top_faved")
+
+            sortOrderFilter?.state == 4 ->
+                sidebarRequest("top_fapped")
+
+            sortOrderFilter?.state == 5 ->
+                sidebarRequest("top_downloaded")
+
+            else ->
+                super.searchMangaRequest(page, query, filters)
+        }
+    }
+
+    private var csrfToken: String? = null
+
+    private fun sidebarRequest(category: String): Request {
+        // Use Elvis operator to fetch CSRF only if it is null
+
+        csrfToken ?: runBlocking {
```

Comment:
Can you get the CSRF token via an interceptor instead? `csrfTokenInterceptor` from WebNovel is an example. There are similar interceptors, but they are named something else.

Reply:
WebNovel intercepts the CSRF token via an header. HentaiFox instead embeds it into the homepage itself, thus i don't think that i can intercept it the same way WebNovel does. I fetch the homepage once and then cache the token, so the traffic rate increase should be negligible

Reply:
The only concern would be the Token's validity period, but in my 15 minutes of testing i have never seen it expire across multiple searches

Reply:
csrf-token is available on other pages as well, the requestTags for example. Maybe we can get that while loading tags?